### PR TITLE
Docker compose for horizontal cromwell

### PR DESCRIPTION
**Pull request by Horneth**
Definitely not for prod but could be useful for dev purposes when one wants to spin up a few Cromwells on the same DB. A load balancer in front routes requests to underlying Cromwells talking to a MySQL instance.

For example:

```
$ docker-compose -f scripts/docker-compose-mysql/docker-compose-cloudwell.yml up --scale cromwell=3 -d
Starting docker-compose-mysql_mysql-db_1 ... done
Starting docker-compose-mysql_cromwell_1 ... done
Starting docker-compose-mysql_cromwell_2 ... done
Starting docker-compose-mysql_cromwell_3 ... done
Starting docker-compose-mysql_lb_1       ... done

$ docker ps
CONTAINER ID        IMAGE                             COMMAND                  CREATED             STATUS                    PORTS                                     NAMES
9311d2f053d6        broadinstitute/cromwell:develop   "/bin/bash -c 'java …"   6 minutes ago       Up 15 seconds             8000/tcp                                  docker-compose-mysql_cromwell_3
e040065fe3ba        broadinstitute/cromwell:develop   "/bin/bash -c 'java …"   6 minutes ago       Up 15 seconds             8000/tcp                                  docker-compose-mysql_cromwell_2
aff5007b8d26        dockercloud/haproxy               "/sbin/tini -- docke…"   7 minutes ago       Up 14 seconds             443/tcp, 1936/tcp, 0.0.0.0:8000->80/tcp   docker-compose-mysql_lb_1
b991a7c412a5        broadinstitute/cromwell:develop   "/bin/bash -c 'java …"   7 minutes ago       Up 15 seconds             8000/tcp                                  docker-compose-mysql_cromwell_1
30892dce18b2        mysql:5.7                         "docker-entrypoint.s…"   17 minutes ago      Up 18 seconds (healthy)   0.0.0.0:3306->3306/tcp                    docker-compose-mysql_mysql-db_1
```

### CODE REVIEW

**Comment by danbills** [comment]
:+1: Glad to see `cloudwell` taking off :)

**Comment by antonkulaga** [comment]
I think that part of the compose file above is not supported by docker swarm

File: src/ci/bin/testDockerScripts.sh [hunk 1]
```diff
@@ -20,3 +20,23 @@ docker run --rm "${docker_tag}" which sbt
 
 echo "2. Testing sbt assembly"
 docker run --rm -v "${PWD}:${PWD}" -w "${PWD}" "${docker_tag}" sbt assembly
+
+echo "Testing cloudwell docker compose"
```
Comment (mcovarr) [inline src/ci/bin/testDockerScripts.sh]: should this be "3. Testing..."?

Comment (Horneth) [reply src/ci/bin/testDockerScripts.sh]: Hmm sure

**Comment by Horneth** [comment]
@antonkulaga any hint on what part specifically ? FWIW this is not intended to be run on a docker swarm but rather provide something to test multi-cromwell on single database. It's really more of "development level" type of script.

**Comment by antonkulaga** [comment]
@Horneth if I remember correctly it can be healthchecks. However, the best way to try is just to init swarm on your laptop and run your yml as a docker stack. Often they just ignore non-supported properties without crashes.
I personally use cromwell in a swarm settings ( https://github.com/antonkulaga/cromwell-client/blob/master/services/pipelines.yml ), however, not froma  real nead but more of convenience, as there is dockerswarmpit with a nice UI to manage services that does not work with compose

**Comment by Horneth** [comment]
@antonkulaga good to know thanks, I knew that they removed health checks in the version 3 of docker compose files but didn't know it would affect swarm. I'll try to give it a go when I have some time.

**Pull request merged**